src/UserService.Business/Commands/Education/CreateEducationCommand.cs

```diff
@@ -1,67 +1,132 @@
 {
-    public class CreateEducationCommand : ICreateEducationCommand
+  public class CreateEducationCommand : ICreateEducationCommand
+  {
+    private readonly IAccessValidator _accessValidator;
+    private readonly IHttpContextAccessor _httpContextAccessor;
+    private readonly IDbUserEducationMapper _mapper;
+    private readonly IUserRepository _userRepository;
+    private readonly IEducationRepository _educationRepository;
+    private readonly ICreateEducationRequestValidator _validator;
+    private readonly IRequestClient<ICreateImagesRequest> _createImagesRequest;
+    private readonly ILogger<CreateEducationCommand> _logger;
+
+    public CreateEducationCommand(
+    IAccessValidator accessValidator,
+    IHttpContextAccessor httpContextAccessor,
+    IDbUserEducationMapper mapper,
+    IUserRepository userRepository,
+    IEducationRepository educationRepository,
+    ICreateEducationRequestValidator validator,
+    IRequestClient<ICreateImagesRequest> createImagesRequest,
+    ILogger<CreateEducationCommand> logger)
     {
-        private readonly IAccessValidator _accessValidator;
-        private readonly IHttpContextAccessor _httpContextAccessor;
-        private readonly IDbUserEducationMapper _mapper;
-        private readonly IUserRepository _userRepository;
-        private readonly IEducationRepository _educationRepository;
-        private readonly ICreateEducationRequestValidator _validator;
-
-        public CreateEducationCommand(
-            IAccessValidator accessValidator,
-            IHttpContextAccessor httpContextAccessor,
-            IDbUserEducationMapper mapper,
-            IUserRepository userRepository,
-            IEducationRepository educationRepository,
-            ICreateEducationRequestValidator validator)
-        {
-            _accessValidator = accessValidator;
-            _httpContextAccessor = httpContextAccessor;
-            _mapper = mapper;
-            _userRepository = userRepository;
-            _educationRepository = educationRepository;
-            _validator = validator;
-        }
+      _accessValidator = accessValidator;
+      _httpContextAccessor = httpContextAccessor;
+      _mapper = mapper;
+      _userRepository = userRepository;
+      _educationRepository = educationRepository;
+      _validator = validator;
+      _createImagesRequest = createImagesRequest;
+      _logger = logger;
+    }
+
+    public OperationResultResponse<Guid> Execute(CreateEducationRequest request)
+    {
+      Guid senderId = _httpContextAccessor.HttpContext.GetUserId();
+      DbUser sender = _userRepository.Get(senderId);
+
+      OperationResultResponse<Guid> response = new();
+
+      if (!(sender.IsAdmin || _accessValidator.HasRights(Rights.AddEditRemoveUsers)) && senderId != request.UserId)
+      {
+        _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.Forbidden;
+
+        response.Errors.Add($"User {senderId} doesn`t have enough rights.");
```

Comment:
do we want to send ids in response errors?

Reply:
summon @MariyaSin

Reply:
just Not enough rights

Reply:
++